### PR TITLE
PWN-5019 - Adding requests for solend main screen

### PR DESCRIPTION
**Pull request by 1sakov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5019

## Description of Work

Using SolendSdk requests for fetching and showing data

## Screenshots (Optional)

Screenshot_20221004_182916

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/sdk/facade/SolendSdkFacade.kt [hunk 1]
```diff
@@ -34,7 +33,8 @@ class SolendSdkFacade(
 ) {
 
     private val currentNetworkEnvironment: NetworkEnvironment
-        get() = networkEnvironmentManager.loadCurrentEnvironment()
+        //        get() = networkEnvironmentManager.loadCurrentEnvironment()
+        get() = NetworkEnvironment.MAINNET
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/sdk/facade/SolendSdkFacade.kt:37]: MAINNET is fixed rn?

File: app/src/main/java/org/p2p/wallet/solend/interactor/SolendDepositsInteractor.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+) {
+
+    suspend fun getAllDeposits(tokenSymbols: List<String>): List<SolendDepositToken> {
+        val marketsInfo = repository.getSolendMarketInfo(tokenSymbols)
+        val activeDeposits = repository.getDeposits()
+
+        return marketsInfo.mapNotNull { info ->
+            val tokenData = userLocalRepository.findTokenDataBySymbol(info.tokenSymbol) ?: return@mapNotNull null
+            val activeDeposit = activeDeposits.find { it.depositTokenSymbol == info.tokenSymbol }
+            mapper.toDepositToken(tokenData, info, activeDeposit)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/solend/interactor/SolendDepositsInteractor.kt:21]: interactor is not responsible for mapping network entities to domain entities
put mapping in the SolendDepositsRepository

File: app/src/main/java/org/p2p/wallet/solend/model/SolendDomainMapper.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package org.p2p.wallet.solend.model
+
+import org.p2p.wallet.user.model.TokenData
+
+class SolendDomainMapper {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/solend/model/SolendDomainMapper.kt:5]: SolendDepositTokenMapper?

File: app/src/main/java/org/p2p/wallet/solend/repository/SolendDepositsRemoteRepository.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    private val ownerAddressProvider: TokenKeyProvider,
+    private val mapper: SolendDepositMapper
+) : SolendDepositsRepository {
+
+    private val currentSolendPool = SolendPool.MAIN
+
+    private val ownerAddress: Base58String
+        get() = ownerAddressProvider.publicKey.toBase58Instance()
+
+    override suspend fun getDeposits(): List<SolendUserDeposit> = try {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/solend/repository/SolendDepositsRemoteRepository.kt:24]: getUserDeposits()

File: app/src/main/java/org/p2p/wallet/solend/ui/earn/SolendEarnPresenter.kt [hunk 5]
```diff
@@ -2,46 +2,45 @@ package org.p2p.wallet.solend.ui.earn
...
                 view?.showLoading(isLoading = false)
             }
         }
     }
+
+    override fun refresh() {
+        view?.showRefreshing(isRefreshing = true)
+        launch {
+            try {
+                val deposits = solendDepositsInteractor.getAllDeposits(COLLATERAL_ACCOUNTS)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/solend/ui/earn/SolendEarnPresenter.kt:36]: maybe COLLATERAL_ACCOUNTS - is a domain entity? it's about domain logic, not presentation

File: app/src/test/java/org/p2p/wallet/sdk/facade/mapper/SolendMethodResultMapperTest.kt [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.intellij.lang.annotations.Language
+import org.junit.Test
+import org.p2p.wallet.sdk.facade.model.SolendMarketInformationResponse
+import org.p2p.wallet.sdk.facade.model.SolendMethodResultException
+
+class SolendMethodResultMapperTest {
```
Comment (gslevinkov) [inline app/src/test/java/org/p2p/wallet/sdk/facade/mapper/SolendMethodResultMapperTest.kt:12]: wow, test master

**Review by gslevinkov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/sdk/facade/SolendSdkFacade.kt (hunk 1 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/sdk/facade/SolendSdkFacade.kt:37]: this is the only one which works for now

File: app/src/main/java/org/p2p/wallet/solend/interactor/SolendDepositsInteractor.kt (hunk 2 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/solend/interactor/SolendDepositsInteractor.kt:21]: This is not network entities, these are merging several domain entities to single domain entity. 
And I was really struggling about the place of mapping. 

I am not sure if we need to make the repository method be complex with several remote requests. 

What do you think guys?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/solend/interactor/SolendDepositsInteractor.kt:21]: i think if domain requests complex domain entity - it's repository responsibility to make all the needed requests and build the entity.
just like with tables - if we want multiple complex data - the repository makes all the requests in database and build it in the mapper

Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/solend/interactor/SolendDepositsInteractor.kt:21]: okay, let me try to move it in repo

File: app/src/main/java/org/p2p/wallet/solend/ui/earn/SolendEarnPresenter.kt (hunk 5 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/solend/ui/earn/SolendEarnPresenter.kt:36]: I think user should know entering presenter what accounts should be visible in that particalar fragment and control it from there. 
So I assume the accounts list is more as a parameter from a user but not as a result of business logic

File: app/src/main/java/org/p2p/wallet/solend/interactor/SolendDepositsInteractor.kt (hunk 2 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/solend/interactor/SolendDepositsInteractor.kt]: done

**Pull request merged**